File: graal-nodejs/test/graal/unit/array.js
/*
 * Copyright (c) 2015, Oracle and/or its affiliates. All rights reserved.
 * ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 */

var assert = require('assert');
var module = require('./_unit');

describe('Array', function () {
    describe('New', function () {
        it('should create an empty array', function () {
            var arr = module.Array_New(0);
            assert.strictEqual(arr instanceof Array, true);
            assert.strictEqual(arr.length, 0);
        });
        it('should create an empty array of certain length', function () {
            var arr = module.Array_New(123);
            assert.strictEqual(arr instanceof Array, true);
            assert.strictEqual(arr.length, 123);
        });
    });
    describe('Length', function () {
        it('should return 0 for []', function () {
            assert.strictEqual(module.Array_Length([]), 0);
        });
        it('should return 3 for [0,1,2]', function () {
            assert.strictEqual(module.Array_Length([0, 1, 2]), 3);
        });
        it('should return 9999 for new Array(9999)', function () {
            assert.strictEqual(module.Array_Length(new Array(9999)), 9999);
        });
    });
});

File: graal-nodejs/test/graal/unit/array.cc
/*
 * Copyright (c) 2015, Oracle and/or its affiliates. All rights reserved.
 * ORACLE PROPRIETARY/CONFIDENTIAL. Use is subject to license terms.
 */

#define SUITE Array

// Array::Length

EXPORT_TO_JS(Length) {
    args.GetReturnValue().Set(args[0].As<Array>()->Length());
}

// Array::New

EXPORT_TO_JS(New) {
    int len = args[0]->ToInteger()->Value();
    args.GetReturnValue().Set(Array::New(args.GetIsolate(), len));
}

#undef SUITE
